### DISCUSSION TITLE
Copy Paste from Google Sheets

### DISCUSSION THREAD

**Discussion by Michael-McAdam** [AntlerVC/firetable#787] [Support / Q&A] [answered]
Is it possible to copy and paste from another cell based editor e.g Google Sheets or Excel?

I tried by it either didn't seem to work, or it pasted it all into a single Rowy cell rather than expanding to the rows and columns correctly.

**Comment by BeeBombshell** [comment, marked as answer]
Hi @Michael-McAdam, rather than copy-pasting the cells directly to a Rowy table, you could simply export the Excel/ Google Sheets file to a `.csv` format. This can be done using the "Import CSV" button on the top right menu.

<img width="960" alt="image" src="https://user-images.githubusercontent.com/53828745/189685641-1dd0d134-5f3b-4401-8788-b658de44a60a.png">

**Reply by Michael-McAdam** [reply]
Hey @BeeBombshell, is this safe to do if you have existing data in the table, i.e, does it append the new cells to the bottom rather than overwriting existing data?

If this does work for adding new rows, sometimes you might want to update content down a column of already existing records. There is a number of ways someone might want to use this functionality outside of adding rows to the bottom.

**Reply by BeeBombshell** [reply]
Hey @Michael-McAdam, yes, importing data doesn't overwrite the existing data in the table. We just need to make sure that the column names are consistent with the already existing ones. A little tip: for this, you could first export your existing table to a `.csv` or `.tsv` file to get the correct format of the table. Thereafter, you can replace the entries with data that you wish to append.

**Reply by jonathanmotif** [reply]
@BeeBombshell I saw your post and thought to mention: I did export my table from Rowy to get the correct table structure and column names. I updated some cells and re-uploaded that CSV through Import. The result was my entire table was duplicated so I'm left wondering how to undo this efficiently. Do you have any idea?

**Reply by BeeBombshell** [reply]
Hi @jonathanmotif, one solution for this would be through sorting the rows by the `createdAt`  field, if you have auditing options enabled for the table. Otherwise, you might need to write a script and handle it with Firebase admin sdk directly, if the number of these rows are high.
